R/utilities-units.R

```diff
@@ -456,19 +456,20 @@ initializeDimensionAndUnitLists <- function() {
 #' @keywords internal
 .unitConverter <- function(data, xUnit = NULL, yUnit = NULL) {
 
-  # target units --------------------------
-
   # No validation of inputs for this non-exported function.
   # All validation will take place in the `DataCombined` class itself.
 
+  # target units --------------------------
+
   # The observed and simulated data should have the same units for
   # visual/graphical comparison.
   #
   # Therefore, if target units are not specified by the user, we need to choose
-  # one ourselves. For no special reason, the first element from a vector of
-  # unique units will be selected: one for X-axis, and one for Y-axis, i.e.
-  xTargetUnit <- xUnit %||% unique(data$xUnit)[[1]]
-  yTargetUnit <- yUnit %||% unique(data$yUnit)[[1]]
+  # one ourselves. The most frequent units will be selected: one for X-axis, and
+  # one for Y-axis. If multiple units are tied in terms of their frequency, the
+  # first will be selected.
+  xTargetUnit <- xUnit %||% .extractMostFrequentUnit(data, unitColumn = xUnit)
```

Comment:
Oh wow, I see. Please let's use strings `""` when these are really strings, otherwise it is super confusing. I always think we are passing the value with a variable. So `unitColumn = "xUnit")